nixos/modules/virtualisation/oci-containers.nix

```diff
@@ -205,24 +208,27 @@ let
     };
 
   mkService = name: container: let
-    mkAfter = map (x: "docker-${x}.service") container.dependsOn;
-  in rec {
+    dependsOn = map (x: "${cfg.backend}-${x}.service") container.dependsOn;
+  in {
     wantedBy = [] ++ optional (container.autoStart) "multi-user.target";
-    after = [ "docker.service" "docker.socket" ] ++ mkAfter;
-    requires = after;
-    path = [ pkgs.docker ];
+    after = lib.optionals (cfg.backend == "docker") [ "docker.service" "docker.socket" ] ++ dependsOn;
+    requires = dependsOn;
+    # Use config.system.path so additional container runtimes in systemPackages are discovered.
+    path = [ config.system.path ];
 
     preStart = ''
-      docker rm -f ${name} || true
+      ${cfg.backend} rm -f ${name} || true
       ${optionalString (container.imageFile != null) ''
-        docker load -i ${container.imageFile}
+        ${cfg.backend} load -i ${container.imageFile}
         ''}
       '';
-    postStop = "docker rm -f ${name} || true";
-        
+    postStop = "${cfg.backend} rm -f ${name} || true";
+
     serviceConfig = {
```

Comment:
In the case of `podman`, do we need to set `environment = config.networking.proxy.envVars` here too?

For the docker backend, this unit only does IPC with the daemon (and the daemon has this already set properly), but for podman, it needs to inherit system-wide proxy servers IIRC.

Reply:
Very good point. Maybe we should simply always set the proxy env vars regardless of backend?

Reply:
SGTM.